server/bgcompute.py

```diff
@@ -32,6 +33,21 @@ def bgcompute_compute_round_contests_sample_sizes():
                 )
             )
 
+            # Claim this RoundContest by updating the sample_size_options field
+            # to an "in progress" value This acts like a lock on the
+            # RoundContest so it can only get processed once (since this
+            # process has a side effect of sampling ballots in some cases,
+            # which we only want to happen once).
+            result = db_session.execute(
+                update(RoundContest.__table__)  # pylint: disable=no-member
+                .where(RoundContest.contest_id == round_contest.contest_id)
+                .where(RoundContest.round_id == round_contest.round_id)
+                .where(RoundContest.sample_size_options.is_(None))
+                .values(sample_size_options="CALCULATION_IN_PROGRESS")
```

Comment:
a better way to do this would be with a `select for update` that would be cleared automatically if an error occurs. I'm okay with this for now, but worth further improvement later.

Reply:
ok, hopefully this code will not exist much longer! it's single-jurisdiction flow stuff